### PR TITLE
dns01 for wildcard portal domain

### PR DESCRIPTION
**Pull request by alexadrake**

### CODE REVIEW

**Review by simplesagar — COMMENTED** [review]
Looks good to me so far

File: charts/speakeasy-k8s/templates/cert-manager.yaml [hunk 1]
```diff
@@ -12,7 +12,34 @@ spec:
     privateKeySecretRef:
       name: speakeasy-private-key
     solvers:
-      - http01:
-          ingress:
-            class: nginx
+    - http01:
+        ingress:
+          class: nginx
+  {{- if index .Values.portal.enabled }}
```
Comment (anuraagnalluri) [inline charts/speakeasy-k8s/templates/cert-manager.yaml:18]: Is index needed here? You're not accessing an array it seems and nothing in the path has dashes (using index seems to be a workaround for that as mentioned here: https://kb.novaordis.com/index.php/Helm_Chart_values.yaml#Field_Names_and_Dashes).

File: charts/speakeasy-k8s/templates/ingress.yaml [hunk 2]
```diff
@@ -75,4 +78,23 @@ spec:
...
+                name: {{ $serviceName }}
+                port:
+                  number: 80
+            path: /v1/
+            pathType: Prefix
+          - backend:
+              service:
+                name: {{ $serviceName }}
+                port:
+                  number: 81
```
Comment (anuraagnalluri) [inline charts/speakeasy-k8s/templates/ingress.yaml:96]: This goes to web, right? Do we need to add another target [here](https://github.com/speakeasy-api/helm-charts/blob/93903fa4f08b5670f4878c34d8e1359175beb9ed/charts/speakeasy-k8s/templates/registry-web-service.yaml) for the portal?

Comment (anuraagnalluri) [reply charts/speakeasy-k8s/templates/ingress.yaml:98]: @alexadrake Reminder to point this at 83

**Review by anuraagnalluri — APPROVED** [review]

File: charts/speakeasy-k8s/templates/ingress.yaml [hunk 3]
```diff
@@ -23,6 +23,9 @@ spec:
         {{- range $v := .Values.registry.ingress.embedFixtureHostnames }}
         - {{ $v }}
         {{- end }}
+        {{- if .Values.portal.enabled }}
+        - {{ .Values.portal.portalWildcardDomain | quote }}
```
Comment (anuraagnalluri) [inline charts/speakeasy-k8s/templates/ingress.yaml:27]: Does this key exist? I only see `portalWildcardDomains` plural

File: charts/speakeasy-k8s/ambassador/ambassador-mappings-and-hosts.yaml [hunk 4]
```diff
@@ -53,6 +53,16 @@ spec:
 ---
 apiVersion: getambassador.io/v3alpha1
 kind: Mapping
+metadata:
+  name: speakeasy-portal
+spec:
+  hostname: $YOUR_PORTAL_HOSTNAME$
+  prefix: /
+  # TODO: point to portal port
+  service: speakeasy-service:81
```
Comment (anuraagnalluri) [inline charts/speakeasy-k8s/ambassador/ambassador-mappings-and-hosts.yaml:62]: Address TODO here as well

File: charts/speakeasy-k8s/templates/ingress.yaml (hunk 3 above)
Comment (alexadrake) [reply charts/speakeasy-k8s/templates/ingress.yaml:27]: good catch

**Pull request merged**